Add detailed information in top-level README (#4)
security-summer-school/template-public: README.md

# SSS TODO_track_name

This is the public repository for the **TODO** track of the Security Summer School (SSS).
It contains information available to participants: session content, slides, activities support files.

> Apart from SSS participants, you can use this content by yourself to learn more about **TODO**.

Reviewer: What are you supposed to replace this **TODO** with (web security, binary exploiting...)? Maybe a list of options in brackets would help the one who uses this template, to me it was unclear.

Author: Gotcha, I will clear that.

## Using the Content

As with all SSS tracks, the **TODO** track consists of multiple sessions.
One session typically covers a particular topic of the track.

> Ideally you would go through sessions orderly, but you can jump ahead to a topic of interest.

Reviewer: ```suggestion
Ideally you would go through the sessions orderly, but you can jump ahead to a topic of interest.
```

Each session has its own folder.

> Access the `README.md` file in each session folder.

Reviewer: ```suggestion
The `README.md` file from a session folder details the topic of the session and includes practical demos of tools and snippets of code.
```
(Or something like that - I didn't like the short, repetitive, separate sentences.)

> It details the topic of the session and includes practical demos of tools and snippets of code.

Reviewer: ```suggestion
```

After reading the session contents, jump to the hands-on activities.
A typical activity will challenge you to solve / exploit a given issue and obtain a flag.
Read the activity description in the session contents and then use the activity support file to get to a solution.
Each activity will develop your practical skills and help you get a better understanding of the session topics.

Activities come in two forms:

* **tutorials** that provide a step-by-step detailing of the solution

> * **challenges** that only state the goal and provide an initial skeleton (and maybe some hints);

Reviewer: ```suggestion
* **challenges** that only state the goal and provide an initial skeleton (and maybe some hints), then the rest is up to you
```
I didn't like the unconsistent use of `;` and `.`

>   the rest is up to you.

Reviewer: ```suggestion
```

  Even so, challenge activity folders store the reference solution for you to take a sneak peak in case you're lost.

## Support

In case you have questions or want to know more about a topic, use the the `Discussions` tab or join the [SSS Discord community](https://discord.gg/qx3RmNEVkB).

## Sessions

> The session for the SSS **TODO** track are:

Reviewer: ```suggestion
The sessions for the SSS **TODO** track are as follows:
```

* [Session 01: Name](session-name-1/README.md)
* TODO
* TODO

### Session Folder Structure

A session folder consists of:

* `README.md`: session contents, written in [GitHub Markdown](https://guides.github.com/features/mastering-markdown/)
* `activities/`: each activity has its own folder with support files (including the reference solution)
* `assets/`: support files for session contents (images, diagrams, code snippets, packet captures etc.)
* `slides/`: session slides

### Activity Folder Structure

An activity folder consists of:

* `public/`: support file for the activity: binary files, helper scripts, packet captures etc.

> * `sol`/: the reference solution of the activity;

Reviewer: ```suggestion
* `sol`/: the reference solution of the activity
```
In the other lists you didn't use a `;`

  Tutorial activities feature a `README.md` file in the solution folder, with a detailing of the solution.

## Contributing

Contributions are welcome.

> See [the contribution guide](CONTRIBUTING.md) on how you could report or fix issues and on how you can improve content.

Reviewer: ```suggestion
See the [contribution guide](CONTRIBUTING.md) on how you could report or fix issues and on how you can improve the content.
```

> Reviewers are requested to follow [the reviewing guide](REVIEWING.md).

Reviewer: ```suggestion
Reviewers are requested to follow the [reviewing guide](REVIEWING.md).
```
